### ISSUE TITLE
obtautomer bug

### ISSUE THREAD

**Issue by jeffjanes** [openbabel/openbabel#322]
If I run obtautomer with the smiles:
CCOC(=O)C1=C(COC)C2=C(NC3=C2C(OC(C)C)=CC=C3)C=N1

It generates two tautomeric structures:
COCc1c(=C(OC)O)ncc2=Nc3c(c12)c(ccc3)OC(C)C
COCc1c(ncc2c1c1c(cccc1[nH]2)OC(C)C)C(=O)OC

But the second of those is missing two double bonds and has 4 extra hydrogens.

This bug exists in HEAD (8adb967c7eb81d05e2158fb) and in older commits as well.

**Comment by baoilleach** [comment]
The input structure has an extra carbon compared to the two outputs so I think you may have mixed things up.

For me, with the input above I get:
CCOC(=c1ncc2=Nc3c(-c2c1COC)c(ccc3)OC(C)C)O
COCc1c(ncc2c1c1c(cccc1[nH]2)OC(C)C)C(=O)OCC

These both have the same number of hydrogens. Please reopen if I've made a mistake.

**Comment by jeffjanes** [comment]
You are correct that I must have given the wrong example input to match output.  I tried the simplify the input as much as possible to reproduce the problem, and must have used a prior version of the input.

However, the output you show has 4 more hydrogens in the first line then the second when passed through obabel again to get the MW.  Apparently the problem is not in the output of the tautomer, but rather the reparsing of that output when used as input.

It does work correctly with the ismy parser, but not the ismi parser.  So the bug is somewhere other than where I thought.

**Comment by baoilleach** [comment]
Right - I see it now:
```
> obabel -:"COCc1c(ncc2c1c1c(cccc1[nH]2)OC(C)C)C(=O)OCC" -:"CCOC(=c1ncc2=Nc3c(-c2c1COC)c(ccc3)OC(C)C)O" --append formula -otxt
C19H22N2O4
C19H26N2O4
```

This is fixed by #1576:
```
C19H22N2O4
C19H22N2O4
```
